client/src/client.rs

```diff
@@ -392,19 +394,20 @@ impl<DB: Database> Client<DB> {
         Ok(())
     }
 
-    pub async fn shutdown(&self) {
+    /// Saves last checkpoint of the node.
+    async fn save_last_checkpoint(&self) {
         let node = self.node.read().await;
-        let checkpoint = if let Some(checkpoint) = node.get_last_checkpoint() {
-            checkpoint
-        } else {
-            return;
+        let _checkpoint = if let Some(_checkpoint) = node.get_last_checkpoint() {
```

Comment:
Why are we declaring this first `_checkpoint` variable here?

Also can we rename the second `_checkpoint` variable to `checkpoint`. The underscore prefix is generally reserved for unused variables.

Reply:
Sorry no reason, I tested some code to compile before and I had a warning but it was before adding code in the block so it must be without the underscore as you say.